src/linker/Linker/MessageContainer.cs

```diff
@@ -66,8 +71,12 @@ public static MessageContainer CreateErrorMessage (string text, int code, string
 		/// <returns>New MessageContainer of 'Warning' category</returns>
 		public static MessageContainer CreateWarningMessage (LinkContext context, string text, int code, MessageOrigin origin, WarnVersion version, string subcategory = MessageSubCategory.None)
 		{
-			if (!(code > 2000 && code <= 6000))
+			if ((Assembly.GetExecutingAssembly () == Assembly.GetCallingAssembly ()) &&
+				!(code > 2000 && code <= 6000))
 				throw new ArgumentException ($"The provided code '{code}' does not fall into the warning category, which is in the range of 2001 to 6000 (inclusive).");
+			else if ((Assembly.GetExecutingAssembly () != Assembly.GetCallingAssembly ()) && code <= 6000)
```

Comment:
I'm a bit worried about perf here. For start let's make sure we only call GetCallingAssembly once in this method. And GetExecutingAssembly is probably slower then typeof(MessageContainer).Assembly which should have the same effect.